# From Fabio M. Di Nitto <fabbione@ubuntu.com>
# Date 27th May 2005
# Source Ubuntu kernel-team baz-archive
#        http://people.u.c/~lamont/Archives/kernel-team@ubuntu.com--2005/
#        kernel-debian--pre1,2--2.6.11.93
#
# * The megaraid legacy driver is around only to support AMI megaraid 1 and 2.
#   All the other controllers are supported (according to the code) by the
#   new megaraid driver:
#   - Add patch drivers-scsi-megaraid_spiltup.dpatch:
#     . Split PCI ID's properly between the 2 drivers.
#     . Allow compilation of both drivers at the same time.
#     . Update Kconfig.megaraid to reflect the new changes in the help.
#     . Rename a few things in the old megaraid driver to avoid possible
#       conflicts with the new drivers (NOTE: there might be more that needs
#       to be changed given that now the 2 modules can be loaded at the same
#       time).

diff -urNad linux-source-2.6.12-2.6.11.90/drivers/scsi/megaraid/Kconfig.megaraid /usr/src/dpatchtemp/dpep.eRg2wK/linux-source-2.6.12-2.6.11.90/drivers/scsi/megaraid/Kconfig.megaraid
--- linux-source-2.6.12-2.6.11.90/drivers/scsi/megaraid/Kconfig.megaraid	2005-03-02 08:37:49.000000000 +0100
+++ /usr/src/dpatchtemp/dpep.eRg2wK/linux-source-2.6.12-2.6.11.90/drivers/scsi/megaraid/Kconfig.megaraid	2005-04-19 08:28:09.521239400 +0200
@@ -64,15 +64,11 @@
 	To compile this driver as a module, choose M here: the
 	module will be called megaraid_mbox
 
-if MEGARAID_NEWGEN=n
 config MEGARAID_LEGACY
 	tristate "LSI Logic Legacy MegaRAID Driver"
 	depends on PCI && SCSI
 	help
-	This driver supports the LSI MegaRAID 418, 428, 438, 466, 762, 490
-	and 467 SCSI host adapters. This driver also support the all U320
-	RAID controllers
+	This driver supports the AMI MEGARAID 1 and 2.
 
 	To compile this driver as a module, choose M here: the
 	module will be called megaraid
-endif
diff -urNad linux-source-2.6.12-2.6.11.90/drivers/scsi/megaraid.c /usr/src/dpatchtemp/dpep.eRg2wK/linux-source-2.6.12-2.6.11.90/drivers/scsi/megaraid.c
--- linux-source-2.6.12-2.6.11.90/drivers/scsi/megaraid.c	2005-03-02 08:37:30.000000000 +0100
+++ /usr/src/dpatchtemp/dpep.eRg2wK/linux-source-2.6.12-2.6.11.90/drivers/scsi/megaraid.c	2005-04-19 08:35:13.889725584 +0200
@@ -4565,7 +4565,7 @@
 };
 
 static int __devinit
-megaraid_probe_one(struct pci_dev *pdev, const struct pci_device_id *id)
+megaraid_legacy_probe_one(struct pci_dev *pdev, const struct pci_device_id *id)
 {
 	struct Scsi_Host *host;
 	adapter_t *adapter;
@@ -5033,37 +5033,25 @@
 }
 
 static struct pci_device_id megaraid_pci_tbl[] = {
-	{PCI_VENDOR_ID_DELL, PCI_DEVICE_ID_DISCOVERY,
-		PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
-	{PCI_VENDOR_ID_DELL, PCI_DEVICE_ID_PERC4_DI,
-		PCI_ANY_ID, PCI_ANY_ID, 0, 0, BOARD_64BIT},
-	{PCI_VENDOR_ID_LSI_LOGIC, PCI_DEVICE_ID_PERC4_QC_VERDE,
-		PCI_ANY_ID, PCI_ANY_ID, 0, 0, BOARD_64BIT},
 	{PCI_VENDOR_ID_AMI, PCI_DEVICE_ID_AMI_MEGARAID,
 		PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
 	{PCI_VENDOR_ID_AMI, PCI_DEVICE_ID_AMI_MEGARAID2,
 		PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
-	{PCI_VENDOR_ID_AMI, PCI_DEVICE_ID_AMI_MEGARAID3,
-		PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
-	{PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_AMI_MEGARAID3,
-		PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
-	{PCI_VENDOR_ID_LSI_LOGIC, PCI_DEVICE_ID_AMI_MEGARAID3,
-		PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
 	{0,}
 };
 MODULE_DEVICE_TABLE(pci, megaraid_pci_tbl);
 
 static struct pci_driver megaraid_pci_driver = {
-	.name		= "megaraid",
+	.name		= "megaraidlegacy",
 	.id_table	= megaraid_pci_tbl,
-	.probe		= megaraid_probe_one,
+	.probe		= megaraid_legacy_probe_one,
 	.remove		= __devexit_p(megaraid_remove_one),
 	.driver		= {
 		.shutdown = megaraid_shutdown,
 	},
 };
 
-static int __init megaraid_init(void)
+static int __init megaraid_legacy_init(void)
 {
 	int error;
 
@@ -5102,7 +5090,7 @@
 	return 0;
 }
 
-static void __exit megaraid_exit(void)
+static void __exit megaraid_legacy_exit(void)
 {
 	/*
 	 * Unregister the character device interface to the driver.
@@ -5116,7 +5104,7 @@
 #endif
 }
 
-module_init(megaraid_init);
-module_exit(megaraid_exit);
+module_init(megaraid_legacy_init);
+module_exit(megaraid_legacy_exit);
 
 /* vi: set ts=8 sw=8 tw=78: */
